example/tests/integration/test_model_resource_name.py

```diff
@@ -61,29 +80,43 @@ def test_resource_name_precendence(self, client):
             'resource_name from model incorrect on list')
 
         # serializer > model
-        serializers.CommentSerializer.Meta.resource_name = "resource_name_from_serializer"
+        monkeypatch.setattr(serializers.CommentSerializer.Meta, 'resource_name', 'resource_name_from_serializer', False)
         response = client.get(reverse("comment-list"))
         data = load_json(response.content)['data'][0]
         assert (data.get('type') == 'resource_name_from_serializer'), (
             'resource_name from serializer incorrect on list')
 
         # view > serializer > model
-        views.CommentViewSet.resource_name = 'resource_name_from_view'
+        monkeypatch.setattr(views.CommentViewSet, 'resource_name', 'resource_name_from_view', False)
         response = client.get(reverse("comment-list"))
         data = load_json(response.content)['data'][0]
         assert (data.get('type') == 'resource_name_from_view'), (
             'resource_name from view incorrect on list')
 
+    def test_model_resource_name_create(self, client):
+        models.Comment.__bases__ += (_PatchedModel,)
+        models.Entry.__bases__ += (_PatchedModel,)
+        response = client.post(reverse("comment-list"),
+                               dump_json(self.create_data),
+                               content_type='application/vnd.api+json')
+
+        assert response.status_code == status.HTTP_201_CREATED
+
+    def test_serializer_resource_name_create(self, client, monkeypatch):
+        monkeypatch.setattr(serializers.CommentSerializer.Meta, 'resource_name', 'renamed_comments', False)
+        monkeypatch.setattr(serializers.EntrySerializer.Meta, 'resource_name', 'renamed_entries', False)
+        self.create_data['data']['type'] = 'renamed_comments'
```

Comment:
I think you should probably deepcopy `self.create_data`. Otherwise, I think you're modifying the class state for all tests.